### PR TITLE
REP-2756 Plain Text Tracing UUID

### PR DESCRIPTION
**Pull request by adrianjgeorge**

### CODE REVIEW

**Comment by adrianjgeorge** [comment]
So far just the config

**Comment by dmnjohns** [comment]
If it's just the attribute, do we want to throw it on the example config as well?

**Comment by dmnjohns** [comment]
Config looks good. 👍

**Comment by adrianjgeorge** [comment]
No, I'd prefer people not use it, so I won't put in the place they are going to copy-pasta

**Comment by adrianjgeorge** [comment]
build in case it's still wonky, https://jenkins.openrepose.org/job/repose-gradle-pull_request/607/

File: repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java [hunk 1]
```diff
@@ -398,6 +400,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                         wrappedRequest.addHeader(CommonHttpHeader.TRACE_GUID.toString(),
                                 TracingHeaderHelper.createTracingHeader(traceGUID, wrappedRequest.getHeader(CommonHttpHeader.VIA.toString())));
                     }
+                    if((currentSystemModel.get().getTracingHeader() != null) &&
```
Comment (dmnjohns) [inline repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java]: The two `Optional` checks could probably be cleaned up a tiny bit. For example, we could do something like:
`Optional<TracingHeader> tracingHeader = Optional.ofNullable(currentSystemModel.get().getTracingHeader())`
`if (tracingHeader.map(th -> th.isEnabled()).orElse(true)) {...}`
`if (tracingHeader.map(th -> th.isSecondaryPlainText()).orElse(false)) {...}`

**Comment by dmnjohns** [comment]
All 5 lines of code look good. I kind of inadvertently code reviewed this because it was so short.

👍

File: repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java (hunk 1 above)
Comment (adrianjgeorge) [inline repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java]: I'm not sure that actually reads cleaner

Comment (dmnjohns) [inline repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java]: I suppose it comes down to what a developer is more used to.

I've grown attached to and familiar with Scala's `Option`s and super don't like seeing the `null` keyword anymore. Also, it was an opportunity to inject more Java 8. 😛

**Comment by adrianjgeorge** [comment]
le sigh, style checks are complaining, i'll figure them out and update

**Comment by adrianjgeorge** [comment]
with fixed style, https://jenkins.openrepose.org/job/repose-gradle-pull_request/612/

File: repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java [hunk 2]
```diff
@@ -78,6 +78,8 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
+import static org.openrepose.commons.utils.http.CommonHttpHeader.REQUEST_ID;
```
Comment (wdschei) [inline repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java]: Should this be made an `_` and just bring them all in?
Then a simple find and replace could clean up a lot of code.

File: repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/tracing/TracingHeaderIncludeSessionIdTest.groovy [hunk 3]
```diff
@@ -347,4 +347,33 @@ class TracingHeaderIncludeSessionIdTest extends ReposeValveTest {
...
+            client_tenantid = 1212
+            client_userid = 1212
+            client_token = UUID.randomUUID().toString()
+            tokenExpiresAt = DateTime.now().plusDays(1)
+        }
+
+        def tracingId = UUID.randomUUID().toString()
+        def sessionId = UUID.randomUUID().toString()
+        def jsonTracingHeader = JsonOutput.toJson([sessionId: sessionId, requestId: tracingId, user: 'a', domain: 'b'])
+        def tracingHeader = Base64.encodeBase64String(jsonTracingHeader.bytes)
```
Comment (wdschei) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/tracing/TracingHeaderIncludeSessionIdTest.groovy]: We should probably force UTF-8 here:
```
jsonTracingHeader.getBytes(Charset.forName("UTF-8"))
```

File: repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/tracing/TracingHeaderPlainTextTest.groovy [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+        reposeLogSearch.cleanLog()
+    }
+
+    def 'Parse externally provided X-Trans-Id header and add the Request ID to the request'() {
+        given:
+        def tracingId = UUID.randomUUID().toString()
+        def sessionId = UUID.randomUUID().toString()
+        def jsonTracingHeader = JsonOutput.toJson([sessionId: sessionId, requestId: tracingId, user: 'a', domain: 'b'])
+        def tracingHeader = Base64.encodeBase64String(jsonTracingHeader.bytes)
```
Comment (wdschei) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/tracing/TracingHeaderPlainTextTest.groovy]: Here too. We could define the Repose common `Charset` somewhere in common and then just use it:
```
private final static Charset CHARSET_UTF8 = Charset.forName("UTF-8")
```

**Comment by wdschei** [comment]
I left some nit-picky comments, but this looks good to me.

**Comment by wdschei** [comment]
https://jenkins.openrepose.org/job/repose-gradle-pull_request/615/

**Pull request merged**